src/spi/mod.rs

```diff
@@ -366,7 +367,7 @@ impl<SPI: InnerSpi, S: SCK, MI: MISO, MO: MOSI> FullDuplex<u8> for Spi<SPI, S, M
             // NOTE(read_volatile) read only 1 byte (the svd2rust API only allows
             // reading a half-word)
             return Ok(unsafe {
-                ptr::read_volatile(&self.spi.dr() as *const _ as *const u8)
+                ptr::read_volatile(self.spi.dr() as *const _ as *const u8)
```

Comment:
Damn I didn't re-check this after refactoring, thanks for noticing, `dr()` already returns reference